src/Servers.jl

```diff
@@ -410,104 +307,79 @@ function handle_connection(f::Function, c::Connection;
     return
 end
 
+"creates an async thread that waits a specified amount of time before closing the connection"
+function check_readtimeout(c, readtimeout, wait_for_timeout)
+    @async while wait_for_timeout[]
+        if inactiveseconds(c) > readtimeout
+            @warn "Connection Timeout: $c"
+            try
+                writeheaders(c.io, Response(408, ["Connection" => "close"]))
+            finally
+                close(c)
+            end
+            break
+        end
+        sleep(8 + rand() * 4)
+    end
+    return
+end
 
 """
-Create a `HTTP.Stream` and parse the Request headers from a `HTTP.Transaction`.
-If there is a parse error, send an error Response.
-Otherwise, execute stream processing function `f`.
+Transaction handler: creates a new Stream for the Transaction, calls startread on it,
+then dispatches the stream to the user-provided handler function. Catches errors on all
+IO operations and closes gracefully if encountered.
 """
-function handle_transaction(f::Function, t::Transaction;
-                            final_transaction::Bool=false,
-                            verbose::Bool=false, kw...)
-
-    request = HTTP.Request()
-    http = Streams.Stream(request, t)
+function handle(h, t::Transaction, last::Bool=false)
```

Comment:
As with `handle_connection` I would really rather keep this as `handle_transaction` rather than have this be a method of the global handle function. If a user gets a hold of a `Transaction` object (e.g. through #282) and tries passing it back to the global `handle` function, it certainly won't have a useful result. `Transaction` is stateful and intimately tied to `Connection`. I would much rather be able to read this code as `handle_connection` calls `handle_transaction` and only have to consider the interaction between those two functions (i.e. not have to image the scenarios and states where some generic recursive call to `handle` ends up back here.

Again, this should have no bearing on the user visible `handle` mechanism right?